### PR TITLE
Reduce memory use when generating pages

### PR DESCRIPTION
**Pull request by sasjo**
The `ReportBuilder` kept a list of pages to generate for no obvious reason. This change removes this list and collects and generates pages in a single pass.

### CODE REVIEW

**Comment by sasjo** [comment]
Updated with changed method signature. The unit tests on internal structure in `ReportBuilderTest` is the reason why the method even needs to exist and have a return value.

File: src/main/java/net/masterthought/cucumber/ReportBuilder.java [hunk 1]
```diff
@@ -149,34 +149,31 @@ private void copyResources(String resourceLocation, String... resources) {
...
+		numPages += generatePage(new StepsOverviewPage(reportResult, configuration));
+		numPages += generatePage(new FailuresOverviewPage(reportResult, configuration));
+
+		if (configuration.isTrendsStatsFile()) {
+			numPages += generatePage(new TrendsOverviewPage(reportResult, configuration, trends));
+		}
+		return numPages;
+	}
+	
+	private int generatePage(AbstractPage page) {
```
Comment (damianszczepanik) [inline src/main/java/net/masterthought/cucumber/ReportBuilder.java]: now I think, this method has no sense as this is wrapper for the single method.

File: src/main/java/net/masterthought/cucumber/ReportBuilder.java [hunk 2]
```diff
@@ -149,34 +149,31 @@ private void copyResources(String resourceLocation, String... resources) {
...
+		page.generatePage();
+		return 1;
```
Comment (damianszczepanik) [inline src/main/java/net/masterthought/cucumber/ReportBuilder.java]: also method that returns constant value is part of code smell and should hot happen

**Comment by sasjo** [comment]
Restructured code and updated tests to verify using output HTML files instead of return values from private functions. Removed a test that was only testing internal (now removed) code. Code coverage should be unchanged in target file.

**Comment by damianszczepanik** [comment]
Thank you for this contribution!

**Pull request merged**